pygeoapi/process/manager/postgresdb_.py

```diff
@@ -0,0 +1,261 @@
+# =================================================================
+#
+# Authors: Francesco Martinelli <francesco.martinelli@ingv.it>
+#
+# Copyright (c) 2024 Francesco Martinelli
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+# Requires postgresql database structure.
+# Import dump:
+# psql -U postgres -h 127.0.0.1 -p 5432 postgres <
+#   tests/data/postgres_manager_full_structure.backup.sql
+
+import json
+import logging
+import psycopg2
+import psycopg2.extras
+
+
+from pathlib import Path
+from typing import Any, Tuple
+
+from pygeoapi.util import JobStatus
+from pygeoapi.process.manager.base import BaseManager
+from pygeoapi.process.base import (
+    JobNotFoundError,
+    JobResultNotFoundError,
+    ProcessorGenericError,
+)
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PostgreSqlDBManager(BaseManager):
+    """PostgreSql Manager"""
+
+    def __init__(self, manager_def: dict):
+        """
+        Initialize object
+
+        :param manager_def: manager definition
+
+        :returns: `pygeoapi.process.manager.postgresdb_.PostgreSqlDBManager`
+        """
+
+        super().__init__(manager_def)
+        self.is_async = True
+        self.supports_subscribing = True
+
+        self.__database_connection_parameters = manager_def['connection']
+        try:
+            # Test connection parameters:
+            test_query = """SELECT version()"""
+            with self.get_db_connection() as conn:
+                with conn.cursor() as cur:
+                    cur.execute(test_query)
+                    cur.fetchone()
+        except (Exception) as ex:
```

Comment:
Suggest:

`except Exception as err`

(`err` is a convention in the codebase)